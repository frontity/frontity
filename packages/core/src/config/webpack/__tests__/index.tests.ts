import * as hash from "hash-it";
import * as path from "path";
import getWebpack from "..";
import type { WebpackConfigs } from "../../../../types";

jest.mock("path");
const mockedPath = path as jest.Mocked<typeof path>;
mockedPath.resolve.mockImplementation((_, ...dirs) => dirs.join("/"));

jest.mock("hash-it");
const mockedHash = hash as jest.Mocked<typeof hash>;
mockedHash.default.mockReturnValue("123");

const babel = {
  development: {
    es5: { presets: ["es5-development"], plugins: [] },
    module: { presets: ["module-development"], plugins: [] },
    server: { presets: ["server-development"], plugins: [] },
  },
  production: {
    es5: { presets: ["es5-production"], plugins: [] },
    module: { presets: ["module-production"], plugins: [] },
    server: { presets: ["server-production"], plugins: [] },
  },
};

const frontity = {
  outDir: "build",
};

const entryPoints = [
  {
    name: "server",
    path: "./build/bundling/entry-points/server.js",
  },
  {
    name: "site-1",
    path: "./build/bundling/entry-points/site-1/client.js",
  },
  {
    name: "site-2",
    path: "./build/bundling/entry-points/site-1/client.js",
  },
];

const removeAbsolutePath = (path: string) =>
  path.replace(/^.*\/node_modules\//, "");

const mapFallbacks = (configs: WebpackConfigs) => {
  Object.keys(configs).forEach((mode: "module" | "es5" | "server") => {
    Object.entries(configs[mode].resolve.fallback).forEach(([key, value]) => {
      configs[mode].resolve.fallback[key] = removeAbsolutePath(value);
    });
  });

  return configs;
};

test("Webpack returns for development", () => {
  let configs = getWebpack({
    mode: "development",
    babel: babel["development"],
    frontity,
    entryPoints,
  });

  configs = mapFallbacks(configs);

  expect(configs).toMatchSnapshot();
});

test("Webpack returns for production", () => {
  let configs = getWebpack({
    mode: "production",
    babel: babel["production"],
    frontity,
    entryPoints,
  });

  configs = mapFallbacks(configs);

  expect(configs).toMatchSnapshot();
});

test("Webpack includes the Bundle Analyzer plugin if specified", () => {
  const { es5, module, server } = getWebpack({
    mode: "production",
    babel: babel["production"],
    frontity,
    entryPoints,
    analyze: true,
  });

  expect(es5.plugins).toMatchSnapshot();
  expect(module.plugins).toMatchSnapshot();
  expect(server.plugins).toMatchSnapshot();
});
