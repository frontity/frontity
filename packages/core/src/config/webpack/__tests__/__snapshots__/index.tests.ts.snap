// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Webpack includes the Bundle Analyzer plugin if specified 1`] = `
Array [
  BundleAnalyzerPlugin {
    "logger": Logger {
      "activeLevels": Set {
        "silent",
      },
    },
    "opts": Object {
      "analyzerHost": "127.0.0.1",
      "analyzerMode": "static",
      "analyzerPort": 8888,
      "defaultSizes": "parsed",
      "excludeAssets": null,
      "generateStatsFile": false,
      "logLevel": "silent",
      "openAnalyzer": false,
      "reportFilename": "../analyze/es5-production.html",
      "reportTitle": [Function],
      "startAnalyzer": true,
      "statsFilename": "stats.json",
      "statsOptions": null,
    },
    "server": null,
  },
  WatchIgnorePlugin {
    "paths": Array [
      /build/,
    ],
  },
  IgnorePlugin {
    "checkIgnore": [Function],
    "options": Object {
      "resourceRegExp": /\\^encoding\\$/,
    },
  },
  LoadablePlugin {
    "compiler": null,
    "handleEmit": [Function],
    "opts": Object {
      "chunkLoadingGlobal": "__LOADABLE_LOADED_CHUNKS__",
      "filename": "../bundling/chunks.es5.json",
      "outputAsset": true,
      "path": undefined,
      "writeToDisk": undefined,
    },
    "writeAssetsFile": [Function],
  },
]
`;

exports[`Webpack includes the Bundle Analyzer plugin if specified 2`] = `
Array [
  BundleAnalyzerPlugin {
    "logger": Logger {
      "activeLevels": Set {
        "silent",
      },
    },
    "opts": Object {
      "analyzerHost": "127.0.0.1",
      "analyzerMode": "static",
      "analyzerPort": 8888,
      "defaultSizes": "parsed",
      "excludeAssets": null,
      "generateStatsFile": false,
      "logLevel": "silent",
      "openAnalyzer": false,
      "reportFilename": "../analyze/module-production.html",
      "reportTitle": [Function],
      "startAnalyzer": true,
      "statsFilename": "stats.json",
      "statsOptions": null,
    },
    "server": null,
  },
  WatchIgnorePlugin {
    "paths": Array [
      /build/,
    ],
  },
  IgnorePlugin {
    "checkIgnore": [Function],
    "options": Object {
      "resourceRegExp": /\\^encoding\\$/,
    },
  },
  LoadablePlugin {
    "compiler": null,
    "handleEmit": [Function],
    "opts": Object {
      "chunkLoadingGlobal": "__LOADABLE_LOADED_CHUNKS__",
      "filename": "../bundling/chunks.module.json",
      "outputAsset": true,
      "path": undefined,
      "writeToDisk": undefined,
    },
    "writeAssetsFile": [Function],
  },
]
`;

exports[`Webpack includes the Bundle Analyzer plugin if specified 3`] = `
Array [
  BundleAnalyzerPlugin {
    "logger": Logger {
      "activeLevels": Set {
        "silent",
      },
    },
    "opts": Object {
      "analyzerHost": "127.0.0.1",
      "analyzerMode": "static",
      "analyzerPort": 8888,
      "defaultSizes": "parsed",
      "excludeAssets": null,
      "generateStatsFile": false,
      "logLevel": "silent",
      "openAnalyzer": false,
      "reportFilename": "analyze/server-production.html",
      "reportTitle": [Function],
      "startAnalyzer": true,
      "statsFilename": "stats.json",
      "statsOptions": null,
    },
    "server": null,
  },
  WatchIgnorePlugin {
    "paths": Array [
      /build/,
    ],
  },
  IgnorePlugin {
    "checkIgnore": [Function],
    "options": Object {
      "resourceRegExp": /\\^encoding\\$/,
    },
  },
  LimitChunkCountPlugin {
    "options": Object {
      "maxChunks": 1,
    },
  },
]
`;

exports[`Webpack returns for development 1`] = `
Object {
  "es5": Object {
    "devtool": "eval-source-map",
    "entry": Object {
      "site-1": Array [
        "webpack-hot-middleware/client",
        "./build/bundling/entry-points/site-1/client.js",
      ],
      "site-2": Array [
        "webpack-hot-middleware/client",
        "./build/bundling/entry-points/site-1/client.js",
      ],
    },
    "externals": Object {
      "he": "{}",
      "node-fetch": "window.fetch",
      "url": "{ URL: window.URL }",
    },
    "mode": "development",
    "module": Object {
      "rules": Array [
        Object {
          "exclude": Array [
            /\\(\\\\/\\|\\\\\\\\\\)core-js\\(\\$\\|\\(\\\\/\\|\\\\\\\\\\)\\)/,
            /\\(\\\\/\\|\\\\\\\\\\)webpack\\(\\$\\|\\(\\\\/\\|\\\\\\\\\\)\\)/,
            /\\(\\\\/\\|\\\\\\\\\\)regenerator-runtime\\(\\$\\|\\(\\\\/\\|\\\\\\\\\\)\\)/,
          ],
          "test": /\\\\\\.\\(j\\|t\\)sx\\?\\$/,
          "use": Object {
            "loader": "babel-loader",
            "options": Object {
              "babelrc": false,
              "cacheDirectory": true,
              "cacheIdentifier": "123",
              "plugins": Array [],
              "presets": Array [
                "es5-development",
              ],
            },
          },
        },
        Object {
          "test": /\\\\\\.\\(png\\|jpe\\?g\\|gif\\|svg\\)\\$/,
          "use": Array [
            Object {
              "loader": "file-loader",
              "options": Object {
                "emitFile": true,
                "name": [Function],
                "outputPath": "images",
              },
            },
          ],
        },
        Object {
          "test": /\\\\\\.css\\$/,
          "use": "raw-loader",
        },
        Object {
          "test": /\\\\\\.\\(woff\\(2\\)\\?\\|ttf\\|eot\\)\\(\\\\\\?v=\\\\d\\+\\\\\\.\\\\d\\+\\\\\\.\\\\d\\+\\)\\?\\$/,
          "use": Object {
            "loader": "url-loader",
            "options": Object {
              "emitFile": true,
              "limit": 8192,
              "name": [Function],
              "outputPath": "fonts",
            },
          },
        },
      ],
    },
    "name": "client",
    "output": Object {
      "chunkFilename": "[name].es5.js",
      "filename": "[name].es5.js",
      "path": "build/static",
    },
    "performance": Object {
      "maxAssetSize": 500000,
      "maxEntrypointSize": 500000,
    },
    "plugins": Array [
      WatchIgnorePlugin {
        "paths": Array [
          /build/,
        ],
      },
      IgnorePlugin {
        "checkIgnore": [Function],
        "options": Object {
          "resourceRegExp": /\\^encoding\\$/,
        },
      },
      HotModuleReplacementPlugin {
        "options": Object {},
      },
      LoadablePlugin {
        "compiler": null,
        "handleEmit": [Function],
        "opts": Object {
          "chunkLoadingGlobal": "__LOADABLE_LOADED_CHUNKS__",
          "filename": "../bundling/chunks.es5.json",
          "outputAsset": true,
          "path": undefined,
          "writeToDisk": undefined,
        },
        "writeAssetsFile": [Function],
      },
    ],
    "resolve": Object {
      "alias": Object {
        "any-promise": "promise-monofill",
        "build": "build",
        "ramda/src": "ramda/es",
      },
      "extensions": Array [
        ".js",
        ".jsx",
        ".ts",
        ".tsx",
      ],
    },
    "stats": Object {
      "all": false,
      "assets": true,
      "colors": true,
      "errorDetails": true,
      "errors": true,
      "excludeAssets": /chunks\\\\\\.\\.\\*\\?\\\\\\.json/,
      "hash": false,
      "warnings": true,
    },
    "target": "web",
  },
  "module": Object {
    "devtool": "eval-source-map",
    "entry": Object {
      "site-1": Array [
        "webpack-hot-middleware/client",
        "./build/bundling/entry-points/site-1/client.js",
      ],
      "site-2": Array [
        "webpack-hot-middleware/client",
        "./build/bundling/entry-points/site-1/client.js",
      ],
    },
    "externals": Object {
      "he": "{}",
      "node-fetch": "window.fetch",
      "url": "{ URL: window.URL }",
    },
    "mode": "development",
    "module": Object {
      "rules": Array [
        Object {
          "exclude": Array [
            /\\(\\\\/\\|\\\\\\\\\\)core-js\\(\\$\\|\\(\\\\/\\|\\\\\\\\\\)\\)/,
            /\\(\\\\/\\|\\\\\\\\\\)webpack\\(\\$\\|\\(\\\\/\\|\\\\\\\\\\)\\)/,
            /\\(\\\\/\\|\\\\\\\\\\)regenerator-runtime\\(\\$\\|\\(\\\\/\\|\\\\\\\\\\)\\)/,
          ],
          "test": /\\\\\\.\\(j\\|t\\)sx\\?\\$/,
          "use": Object {
            "loader": "babel-loader",
            "options": Object {
              "babelrc": false,
              "cacheDirectory": true,
              "cacheIdentifier": "123",
              "plugins": Array [],
              "presets": Array [
                "module-development",
              ],
            },
          },
        },
        Object {
          "test": /\\\\\\.\\(png\\|jpe\\?g\\|gif\\|svg\\)\\$/,
          "use": Array [
            Object {
              "loader": "file-loader",
              "options": Object {
                "emitFile": true,
                "name": [Function],
                "outputPath": "images",
              },
            },
          ],
        },
        Object {
          "test": /\\\\\\.css\\$/,
          "use": "raw-loader",
        },
        Object {
          "test": /\\\\\\.\\(woff\\(2\\)\\?\\|ttf\\|eot\\)\\(\\\\\\?v=\\\\d\\+\\\\\\.\\\\d\\+\\\\\\.\\\\d\\+\\)\\?\\$/,
          "use": Object {
            "loader": "url-loader",
            "options": Object {
              "emitFile": true,
              "limit": 8192,
              "name": [Function],
              "outputPath": "fonts",
            },
          },
        },
      ],
    },
    "name": "client",
    "output": Object {
      "chunkFilename": "[name].module.js",
      "filename": "[name].module.js",
      "path": "build/static",
    },
    "performance": Object {
      "maxAssetSize": 500000,
      "maxEntrypointSize": 500000,
    },
    "plugins": Array [
      WatchIgnorePlugin {
        "paths": Array [
          /build/,
        ],
      },
      IgnorePlugin {
        "checkIgnore": [Function],
        "options": Object {
          "resourceRegExp": /\\^encoding\\$/,
        },
      },
      HotModuleReplacementPlugin {
        "options": Object {},
      },
      LoadablePlugin {
        "compiler": null,
        "handleEmit": [Function],
        "opts": Object {
          "chunkLoadingGlobal": "__LOADABLE_LOADED_CHUNKS__",
          "filename": "../bundling/chunks.module.json",
          "outputAsset": true,
          "path": undefined,
          "writeToDisk": undefined,
        },
        "writeAssetsFile": [Function],
      },
    ],
    "resolve": Object {
      "alias": Object {
        "any-promise": "promise-monofill",
        "build": "build",
        "ramda/src": "ramda/es",
      },
      "extensions": Array [
        ".js",
        ".jsx",
        ".ts",
        ".tsx",
      ],
    },
    "stats": Object {
      "all": false,
      "assets": true,
      "colors": true,
      "errorDetails": true,
      "errors": true,
      "excludeAssets": /chunks\\\\\\.\\.\\*\\?\\\\\\.json/,
      "hash": false,
      "warnings": true,
    },
    "target": "web",
  },
  "server": Object {
    "devtool": "eval-source-map",
    "entry": "./build/bundling/entry-points/server.js",
    "externals": Object {},
    "mode": "development",
    "module": Object {
      "rules": Array [
        Object {
          "exclude": Array [
            /\\(\\\\/\\|\\\\\\\\\\)core-js\\(\\$\\|\\(\\\\/\\|\\\\\\\\\\)\\)/,
            /\\(\\\\/\\|\\\\\\\\\\)webpack\\(\\$\\|\\(\\\\/\\|\\\\\\\\\\)\\)/,
            /\\(\\\\/\\|\\\\\\\\\\)regenerator-runtime\\(\\$\\|\\(\\\\/\\|\\\\\\\\\\)\\)/,
          ],
          "test": /\\\\\\.\\(j\\|t\\)sx\\?\\$/,
          "use": Object {
            "loader": "babel-loader",
            "options": Object {
              "babelrc": false,
              "cacheDirectory": true,
              "cacheIdentifier": "123",
              "plugins": Array [],
              "presets": Array [
                "server-development",
              ],
            },
          },
        },
        Object {
          "test": /\\\\\\.\\(png\\|jpe\\?g\\|gif\\|svg\\)\\$/,
          "use": Array [
            Object {
              "loader": "file-loader",
              "options": Object {
                "emitFile": false,
                "name": [Function],
                "outputPath": "images",
              },
            },
          ],
        },
        Object {
          "test": /\\\\\\.css\\$/,
          "use": "raw-loader",
        },
        Object {
          "test": /\\\\\\.\\(woff\\(2\\)\\?\\|ttf\\|eot\\)\\(\\\\\\?v=\\\\d\\+\\\\\\.\\\\d\\+\\\\\\.\\\\d\\+\\)\\?\\$/,
          "use": Object {
            "loader": "url-loader",
            "options": Object {
              "emitFile": false,
              "limit": 8192,
              "name": [Function],
              "outputPath": "fonts",
            },
          },
        },
      ],
    },
    "name": "server",
    "output": Object {
      "filename": "server.js",
      "libraryTarget": "commonjs2",
      "path": "build/",
    },
    "performance": Object {
      "maxAssetSize": 5000000,
      "maxEntrypointSize": 5000000,
    },
    "plugins": Array [
      WatchIgnorePlugin {
        "paths": Array [
          /build/,
        ],
      },
      IgnorePlugin {
        "checkIgnore": [Function],
        "options": Object {
          "resourceRegExp": /\\^encoding\\$/,
        },
      },
      LimitChunkCountPlugin {
        "options": Object {
          "maxChunks": 1,
        },
      },
    ],
    "resolve": Object {
      "alias": Object {
        "any-promise": "promise-monofill",
        "build": "build",
        "ramda/src": "ramda/es",
      },
      "extensions": Array [
        ".js",
        ".jsx",
        ".ts",
        ".tsx",
      ],
    },
    "stats": Object {
      "all": false,
      "assets": true,
      "colors": true,
      "errorDetails": true,
      "errors": true,
      "excludeAssets": /chunks\\\\\\.\\.\\*\\?\\\\\\.json/,
      "hash": false,
      "warnings": true,
    },
    "target": "node",
  },
}
`;

exports[`Webpack returns for production 1`] = `
Object {
  "es5": Object {
    "devtool": false,
    "entry": Object {
      "site-1": Array [
        "./build/bundling/entry-points/site-1/client.js",
      ],
      "site-2": Array [
        "./build/bundling/entry-points/site-1/client.js",
      ],
    },
    "externals": Object {
      "he": "{}",
      "node-fetch": "window.fetch",
      "url": "{ URL: window.URL }",
    },
    "mode": "production",
    "module": Object {
      "rules": Array [
        Object {
          "exclude": Array [
            /\\(\\\\/\\|\\\\\\\\\\)core-js\\(\\$\\|\\(\\\\/\\|\\\\\\\\\\)\\)/,
            /\\(\\\\/\\|\\\\\\\\\\)webpack\\(\\$\\|\\(\\\\/\\|\\\\\\\\\\)\\)/,
            /\\(\\\\/\\|\\\\\\\\\\)regenerator-runtime\\(\\$\\|\\(\\\\/\\|\\\\\\\\\\)\\)/,
          ],
          "test": /\\\\\\.\\(j\\|t\\)sx\\?\\$/,
          "use": Object {
            "loader": "babel-loader",
            "options": Object {
              "babelrc": false,
              "cacheDirectory": true,
              "cacheIdentifier": "123",
              "plugins": Array [],
              "presets": Array [
                "es5-production",
              ],
            },
          },
        },
        Object {
          "test": /\\\\\\.\\(png\\|jpe\\?g\\|gif\\|svg\\)\\$/,
          "use": Array [
            Object {
              "loader": "file-loader",
              "options": Object {
                "emitFile": true,
                "name": [Function],
                "outputPath": "images",
              },
            },
          ],
        },
        Object {
          "test": /\\\\\\.css\\$/,
          "use": "raw-loader",
        },
        Object {
          "test": /\\\\\\.\\(woff\\(2\\)\\?\\|ttf\\|eot\\)\\(\\\\\\?v=\\\\d\\+\\\\\\.\\\\d\\+\\\\\\.\\\\d\\+\\)\\?\\$/,
          "use": Object {
            "loader": "url-loader",
            "options": Object {
              "emitFile": true,
              "limit": 8192,
              "name": [Function],
              "outputPath": "fonts",
            },
          },
        },
      ],
    },
    "name": "client",
    "output": Object {
      "chunkFilename": "[name].es5.[chunkhash].js",
      "filename": "[name].es5.[chunkhash].js",
      "path": "build/static",
    },
    "performance": Object {
      "maxAssetSize": 500000,
      "maxEntrypointSize": 500000,
    },
    "plugins": Array [
      WatchIgnorePlugin {
        "paths": Array [
          /build/,
        ],
      },
      IgnorePlugin {
        "checkIgnore": [Function],
        "options": Object {
          "resourceRegExp": /\\^encoding\\$/,
        },
      },
      LoadablePlugin {
        "compiler": null,
        "handleEmit": [Function],
        "opts": Object {
          "chunkLoadingGlobal": "__LOADABLE_LOADED_CHUNKS__",
          "filename": "../bundling/chunks.es5.json",
          "outputAsset": true,
          "path": undefined,
          "writeToDisk": undefined,
        },
        "writeAssetsFile": [Function],
      },
    ],
    "resolve": Object {
      "alias": Object {
        "any-promise": "promise-monofill",
        "build": "build",
        "ramda/src": "ramda/es",
      },
      "extensions": Array [
        ".js",
        ".jsx",
        ".ts",
        ".tsx",
      ],
    },
    "stats": false,
    "target": "web",
  },
  "module": Object {
    "devtool": false,
    "entry": Object {
      "site-1": Array [
        "./build/bundling/entry-points/site-1/client.js",
      ],
      "site-2": Array [
        "./build/bundling/entry-points/site-1/client.js",
      ],
    },
    "externals": Object {
      "he": "{}",
      "node-fetch": "window.fetch",
      "url": "{ URL: window.URL }",
    },
    "mode": "production",
    "module": Object {
      "rules": Array [
        Object {
          "exclude": Array [
            /\\(\\\\/\\|\\\\\\\\\\)core-js\\(\\$\\|\\(\\\\/\\|\\\\\\\\\\)\\)/,
            /\\(\\\\/\\|\\\\\\\\\\)webpack\\(\\$\\|\\(\\\\/\\|\\\\\\\\\\)\\)/,
            /\\(\\\\/\\|\\\\\\\\\\)regenerator-runtime\\(\\$\\|\\(\\\\/\\|\\\\\\\\\\)\\)/,
          ],
          "test": /\\\\\\.\\(j\\|t\\)sx\\?\\$/,
          "use": Object {
            "loader": "babel-loader",
            "options": Object {
              "babelrc": false,
              "cacheDirectory": true,
              "cacheIdentifier": "123",
              "plugins": Array [],
              "presets": Array [
                "module-production",
              ],
            },
          },
        },
        Object {
          "test": /\\\\\\.\\(png\\|jpe\\?g\\|gif\\|svg\\)\\$/,
          "use": Array [
            Object {
              "loader": "file-loader",
              "options": Object {
                "emitFile": true,
                "name": [Function],
                "outputPath": "images",
              },
            },
          ],
        },
        Object {
          "test": /\\\\\\.css\\$/,
          "use": "raw-loader",
        },
        Object {
          "test": /\\\\\\.\\(woff\\(2\\)\\?\\|ttf\\|eot\\)\\(\\\\\\?v=\\\\d\\+\\\\\\.\\\\d\\+\\\\\\.\\\\d\\+\\)\\?\\$/,
          "use": Object {
            "loader": "url-loader",
            "options": Object {
              "emitFile": true,
              "limit": 8192,
              "name": [Function],
              "outputPath": "fonts",
            },
          },
        },
      ],
    },
    "name": "client",
    "output": Object {
      "chunkFilename": "[name].module.[chunkhash].js",
      "filename": "[name].module.[chunkhash].js",
      "path": "build/static",
    },
    "performance": Object {
      "maxAssetSize": 500000,
      "maxEntrypointSize": 500000,
    },
    "plugins": Array [
      WatchIgnorePlugin {
        "paths": Array [
          /build/,
        ],
      },
      IgnorePlugin {
        "checkIgnore": [Function],
        "options": Object {
          "resourceRegExp": /\\^encoding\\$/,
        },
      },
      LoadablePlugin {
        "compiler": null,
        "handleEmit": [Function],
        "opts": Object {
          "chunkLoadingGlobal": "__LOADABLE_LOADED_CHUNKS__",
          "filename": "../bundling/chunks.module.json",
          "outputAsset": true,
          "path": undefined,
          "writeToDisk": undefined,
        },
        "writeAssetsFile": [Function],
      },
    ],
    "resolve": Object {
      "alias": Object {
        "any-promise": "promise-monofill",
        "build": "build",
        "ramda/src": "ramda/es",
      },
      "extensions": Array [
        ".js",
        ".jsx",
        ".ts",
        ".tsx",
      ],
    },
    "stats": false,
    "target": "web",
  },
  "server": Object {
    "devtool": false,
    "entry": "./build/bundling/entry-points/server.js",
    "externals": Object {},
    "mode": "production",
    "module": Object {
      "rules": Array [
        Object {
          "exclude": Array [
            /\\(\\\\/\\|\\\\\\\\\\)core-js\\(\\$\\|\\(\\\\/\\|\\\\\\\\\\)\\)/,
            /\\(\\\\/\\|\\\\\\\\\\)webpack\\(\\$\\|\\(\\\\/\\|\\\\\\\\\\)\\)/,
            /\\(\\\\/\\|\\\\\\\\\\)regenerator-runtime\\(\\$\\|\\(\\\\/\\|\\\\\\\\\\)\\)/,
          ],
          "test": /\\\\\\.\\(j\\|t\\)sx\\?\\$/,
          "use": Object {
            "loader": "babel-loader",
            "options": Object {
              "babelrc": false,
              "cacheDirectory": true,
              "cacheIdentifier": "123",
              "plugins": Array [],
              "presets": Array [
                "server-production",
              ],
            },
          },
        },
        Object {
          "test": /\\\\\\.\\(png\\|jpe\\?g\\|gif\\|svg\\)\\$/,
          "use": Array [
            Object {
              "loader": "file-loader",
              "options": Object {
                "emitFile": false,
                "name": [Function],
                "outputPath": "images",
              },
            },
          ],
        },
        Object {
          "test": /\\\\\\.css\\$/,
          "use": "raw-loader",
        },
        Object {
          "test": /\\\\\\.\\(woff\\(2\\)\\?\\|ttf\\|eot\\)\\(\\\\\\?v=\\\\d\\+\\\\\\.\\\\d\\+\\\\\\.\\\\d\\+\\)\\?\\$/,
          "use": Object {
            "loader": "url-loader",
            "options": Object {
              "emitFile": false,
              "limit": 8192,
              "name": [Function],
              "outputPath": "fonts",
            },
          },
        },
      ],
    },
    "name": "server",
    "output": Object {
      "filename": "server.js",
      "libraryTarget": "commonjs2",
      "path": "build/",
    },
    "performance": Object {
      "maxAssetSize": 5000000,
      "maxEntrypointSize": 5000000,
    },
    "plugins": Array [
      WatchIgnorePlugin {
        "paths": Array [
          /build/,
        ],
      },
      IgnorePlugin {
        "checkIgnore": [Function],
        "options": Object {
          "resourceRegExp": /\\^encoding\\$/,
        },
      },
      LimitChunkCountPlugin {
        "options": Object {
          "maxChunks": 1,
        },
      },
    ],
    "resolve": Object {
      "alias": Object {
        "any-promise": "promise-monofill",
        "build": "build",
        "ramda/src": "ramda/es",
      },
      "extensions": Array [
        ".js",
        ".jsx",
        ".ts",
        ".tsx",
      ],
    },
    "stats": false,
    "target": "node",
  },
}
`;
